### PR TITLE
Add support for METAFONT

### PR DESCRIPTION
**Pull request by LaeriExNihilo**
Includes support for Donald Knuth's historic language for fontdesigning and fontmaking. While this language is outdated in many regards its legacy endures in METAPOST and PGF/TikZ, and its macro core principle has real depth. May be of use for people exploring the inception of the TeX ecosystem.

### CODE REVIEW

File: components/prism-metafont.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    'type': {
+        pattern: /\b(?:boolean|expr|numeric|pair|path|pen|picture|primary|secondary|string|suffix|tertiary|text|transform)\b/,
+        alias: 'property'
+    },
+    'variable': {
+        pattern: /(^|[^@#&$])(?:@#|#@|#|@)(?![@#&$])|\b(?:aspect_ratio|currentpen|currentpicture|currenttransform|d|extra_beginchar|extra_endchar|extra_setup|h|localfont|mag|mode|screen_cols|screen_rows|whatever|w|x|y|z)\b/,
+        lookbehind: true
+    }
+};
+Prism.languages.mf=Prism.languages['metafont'];
```
Comment (RunDevelopment) [inline components/prism-metafont.js:85]: Could you please remove the alias (here and in `components.json`)?

Short aliases are limited resource, and I don't like the idea of `mf` being taken by a language that basically isn't used anymore.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you for the PR @LaeriExNihilo!

The language itself looks good, but there aren't enough tests, so our regex coverage fails. You can see the untested regexes/alternatives [here](https://github.com/PrismJS/prism/runs/6752912791?check_suite_focus=true#step:5:1267) or by running `npm run regex-coverage`.

**Comment by LaeriExNihilo** [comment]
This should pass now @RunDevelopment . While doing the tests I also found a few mistakes in my definitions so it was a good opportunity.

**Comment by RunDevelopment** [comment]
Thanks @LaeriExNihilo! I forgot to mention that we also have ESLint. Please run `npm run lint:fix` to fix (hopefully) all linting errors.

**Comment by LaeriExNihilo** [comment]
lint is complaining about a duplicate for my operators but I can't seem to find it. I fixed the spacing (that VSCode had by default to spaces instead of tabs) but I'm not sure it will fix it.

**Comment by LaeriExNihilo** [comment]
Locally this build passes all tests. lint fixed some alphabetical misorderings I hadn't seen.

**Comment by RunDevelopment** [comment]
Thank you for contributing @LaeriExNihilo!

**Pull request merged**